src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala

```diff
@@ -30,6 +30,23 @@ class Tile(var tileId: TileId, var latitudeNorth: Double, var latitudeSouth: Dou
   def midSouthLatitude(): Double = (centerLatitude() + latitudeSouth) / 2.0
   def midEastLongitude(): Double = (centerLongitude() + longitudeEast) / 2.0
   def midWestLongitude(): Double = (centerLongitude() + longitudeWest) / 2.0
+
+  override def toString = s"Tile($tileId, $latitudeNorth, $latitudeSouth, $longitudeWest, $longitudeEast)"
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[Tile]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Tile =>
+      (that canEqual this) &&
+        tileId == that.tileId
+    case _ => false
+  }
```

Comment:
Making sure this returns true for equal values of `tileId` regardless of lat/long values (and precision).